pkg/util/winutil/etw/etw.go

```diff
@@ -76,7 +110,7 @@ func isHTTPServiceSubscriptionEnabled(etwProviders ProviderType) bool {
 func etwCallbackC(eventInfo *C.DD_ETW_EVENT_INFO) {
 	switch eventInfo.provider {
 	case C.DD_ETW_TRACE_PROVIDER_HttpService:
-		if sub, ok := subscribers[EtwProviderHTTPService]; ok {
+		if sub, ok := safeSubscribers.Get(EtwProviderHTTPService); ok {
```

Comment:
The access themselves are now safe, but there is no guaranty that the item is still in the map when you'll access it.

For this kind of logic it would probably be safer to let the locking be handled by the caller, or to perform a deep copy of the extracted item

Reply:
That is true, it should have a `Stop` method on the map to lock it